tests/rptest/services/keycloak.py

```diff
@@ -0,0 +1,237 @@
+import json
+import os
+import requests
+
+from ducktape.services.service import Service
+from ducktape.utils.util import wait_until
+
+from keycloak import KeycloakAdmin
+
+KC_INSTALL_DIR = os.path.join('/', 'opt', 'keycloak')
+KC_DATA_DIR = os.path.join(KC_INSTALL_DIR, 'data')
+KC_VAULT_DIR = os.path.join(KC_DATA_DIR, 'vault')
+KC_BIN_DIR = os.path.join(KC_INSTALL_DIR, 'bin')
+KC = os.path.join(KC_BIN_DIR, 'kc.sh')
+KCADM = os.path.join(KC_BIN_DIR, 'kcadm.sh')
+KC_ADMIN = 'admin'
+KC_ADMIN_PASSWORD = 'admin'
+KC_ROOT_LOG_LEVEL = 'INFO'
+KC_LOG_HANDLER = 'console,file'
+KC_LOG_FILE = '/var/log/kc.log'
+KC_PORT = 8080
+
+DEFAULT_REALM = 'demorealm'
+
+START_CMD_TMPL = """
+LAUNCH_JBOSS_IN_BACKGROUND=1 \
+KEYCLOAK_ADMIN={admin} \
+KEYCLOAK_ADMIN_PASSWORD={pw} \
+{kc} start-dev --http-port={port} \
+--log="{log_handler}" --log-file="{logfile}" --log-level="{log_level}" &
+"""
+
+OIDC_CONFIG_TMPL = """\
+http://{host}:{port}/realms/{realm}/.well-known/openid-configuration\
+"""
+
+
+class OAuthConfig:
+    def __init__(self,
+                 client_id,
+                 client_secret,
+                 token_endpoint,
+                 scopes=['openid']):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.token_endpoint = token_endpoint
+        self.scopes = scopes
+
+
+class KeycloakAdminClient:
+    def __init__(self,
+                 logger,
+                 server_url,
+                 realm,
+                 username=KC_ADMIN,
+                 password=KC_ADMIN_PASSWORD):
+        self.logger = logger
+        self.kc_admin = KeycloakAdmin(
+            server_url=server_url,
+            username=username,
+            password=password,
+            realm_name='master',
+        )
+        self.kc_admin.create_realm(payload={
+            'realm': realm,
+            'enabled': True,
+        })
+        self.kc_admin.realm_name = realm
+
+    def config(self, server_url, username, password, realm):
+        self.kc_admin = KeycloakAdmin(server_url=server_url,
+                                      username=username,
+                                      password=password,
+                                      realm_name=realm)
+
+    def create_client(self, client_id, **kwargs):
+        rep = {
+            'clientId': client_id,
+            'enabled': True,
+            'serviceAccountsEnabled': True,  # for client credentials grant
+            # 'standardFlowEnabled': True,  # authorization flow grant
+            # 'directAccessGrantsEnabled': True, # resource owner password grant
+            # 'implicitFlowEnabled': True,  # implicit grant
```

Comment:
nit: Any reason these should remain here?  For code clarity can we remove commented out code.

Reply:
These are the flags to enable different grant types on the client; left them in for pseudo-documentation purposes because they're slightly obtuse. PRD calls exclusively for `client-credentials` though; will remove.